mypy/checker.py

```diff
@@ -3223,3 +3233,16 @@ def push_class(self, info: TypeInfo) -> Iterator[None]:
 @contextmanager
 def nothing() -> Iterator[None]:
     yield
+
+
+def is_typed_callable(c: Optional[Type]) -> bool:
+    if not c or not isinstance(c, CallableType):
+        return False
+    return not any(isinstance(t, AnyType) and t.implicit
```

Comment:
It looks like this is checking for a fully typed callable, but we should check for a callable with that has _any_ annotations.  This case should have appeared in tests.